sync-core/src/test/java/com/cloudant/sync/query/QueryExecutorTest.java

```diff
@@ -0,0 +1,351 @@
+    public void canQueryOverOneNumberFieldNormalized() {
+        // query - { "age" : { "eq" : 12 } }
+        Map<String, Object> operator = new HashMap<String, Object>();
+        operator.put("$eq", 12);
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("age", operator);
+        QueryResult queryResult = im.find(query);
+        Assert.assertNotNull(queryResult);
+        Assert.assertEquals(2, queryResult.size());
+        List<String> checkList = new ArrayList<String>();
+        checkList.add("mike12");
+        checkList.add("fred12");
+        Assert.assertEquals(2, queryResult.documentIds().size());
+        Assert.assertTrue(queryResult.documentIds().containsAll(checkList));
+        List<String> docCheckList = new ArrayList<String>();
+        for (DocumentRevision rev: queryResult) {
+            Assert.assertNotNull(rev.getId());
+            Assert.assertNotNull(rev.getBody());
+            docCheckList.add(rev.getId());
+        }
+        Assert.assertEquals(2, docCheckList.size());
+        Assert.assertTrue(docCheckList.containsAll(checkList));
+    }
+
+    @Test
+    public void canQueryOverTwoStringFields() {
+        // query - { "name" : "mike", "pet" : "cat" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        query.put("pet", "cat");
+        QueryResult queryResult = im.find(query);
+        Assert.assertNotNull(queryResult);
+        Assert.assertEquals(2, queryResult.size());
+        List<String> checkList = new ArrayList<String>();
+        checkList.add("mike12");
+        checkList.add("mike72");
+        Assert.assertEquals(2, queryResult.documentIds().size());
+        Assert.assertTrue(queryResult.documentIds().containsAll(checkList));
+        List<String> docCheckList = new ArrayList<String>();
+        for (DocumentRevision rev: queryResult) {
+            Assert.assertNotNull(rev.getId());
+            Assert.assertNotNull(rev.getBody());
+            docCheckList.add(rev.getId());
+        }
+        Assert.assertEquals(2, docCheckList.size());
+        Assert.assertTrue(docCheckList.containsAll(checkList));
+    }
+
+    @Test
+    public void canQueryOverTwoStringFieldsNormalized() {
+        // query - { "name" : { "eq" : "mike" }, "pet" : { "$eq" : "cat" } }
+        Map<String, Object> nameOperator = new HashMap<String, Object>();
+        nameOperator.put("$eq", "mike");
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", nameOperator);
+        Map<String, Object> petOperator = new HashMap<String, Object>();
+        petOperator.put("$eq", "cat");
+        query.put("pet", petOperator);
+        QueryResult queryResult = im.find(query);
+        Assert.assertNotNull(queryResult);
+        Assert.assertEquals(2, queryResult.size());
+        List<String> checkList = new ArrayList<String>();
+        checkList.add("mike12");
+        checkList.add("mike72");
```

Comment:
This would seem clearer as `List<String> checkList = Arrays.asList("mike12", "mike72");`